packages/amplify-graphql-api-construct/src/__tests__/__functional__/disable-gen1-patterns.test.ts

```diff
@@ -0,0 +1,353 @@
+import * as cdk from 'aws-cdk-lib';
+import { Template } from 'aws-cdk-lib/assertions';
+import { Bucket } from 'aws-cdk-lib/aws-s3';
+import { AmplifyGraphqlApi } from '../../amplify-graphql-api';
+import { AmplifyGraphqlDefinition } from '../../amplify-graphql-definition';
+
+/**
+ * Utility to test if schema is valid when gen 1 patterns are disabled
+ * @param schema schema to test
+ * @param allowGen1Patterns if gen 1 patterns are allowed.
+ */
+const verifySchema = (schema: string, allowGen1Patterns: boolean): void => {
+  const stack = new cdk.Stack();
+  new AmplifyGraphqlApi(stack, 'TestApi', {
+    definition: AmplifyGraphqlDefinition.fromString(schema),
+    authorizationModes: {
+      apiKeyConfig: { expires: cdk.Duration.days(7) },
+    },
+    translationBehavior: {
+      allowGen1Patterns,
+    },
+  });
+  Template.fromStack(stack);
+};
+
+describe('allowGen1Patterns', () => {
+  test('defaults to allow', () => {
+    const schema = `
+      type Post @model {
+        tags: [Tag] @manyToMany(relationName: "PostTags")
+      }
+
+      type Tag @model {
+        posts: [Post] @manyToMany(relationName: "PostTags")
+      }
+    `;
+    const stack = new cdk.Stack();
+    expect(
+      () =>
+        new AmplifyGraphqlApi(stack, 'TestApi', {
+          definition: AmplifyGraphqlDefinition.fromString(schema),
+          authorizationModes: {
+            apiKeyConfig: { expires: cdk.Duration.days(7) },
+          },
+        }),
+    ).not.toThrow();
+  });
+
+  test('does not allow @manyToMany', () => {
```

Comment:
nit here & throughout: test titles should reflect the conditions better, as in
```suggestion
  test('does not allow @manyToMany when allowGen1Patterns is false', () => {
```

You could accomplish the same thing by organizing nested describe blocks:
```typescript
describe('allowGen1Patterns', () => {
  test('defaults to allow', () => {...});
  describe('flag is true', () => {/* all the cases */});
  describe('flag is false', () => {/* all the cases */});
});